src/main/java/com/opentable/kafka/util/OffsetMonitor.java

```diff
@@ -0,0 +1,137 @@
+        return AdminClient.create(props);
+    }
+
+    @Override
+    public void close() {
+        consumer.close();
+        adminClient.close();
+    }
+
+    /**
+     * Find size (maximum offset) for each partition on topic.
+     *
+     * <p>
+     * If the topic does not exist, the returned map will be empty.
+     *
+     * @param topic the topic for which to fetch partition sizes (maximum offsets)
+     * @return map of partition of size (maximum offset)
+     */
+    public Map<Integer, Long> getTopicSizes(final String topic) {
+        final Collection<PartitionInfo> partInfo = consumer.partitionsFor(topic);
+        if (partInfo == null) {
+            // Topic does not exist.
+            return Collections.emptyMap();
+        }
+        final Collection<TopicPartition> parts = partInfo
+                .stream()
+                .map(pi -> new TopicPartition(pi.topic(), pi.partition()))
+                .collect(Collectors.toSet());
+        final Map<TopicPartition, Long> sizes = consumer.endOffsets(parts);
+        return sizes
+                .entrySet()
+                .stream()
+                .collect(
+                        Collectors.toMap(
+                                entry -> entry.getKey().partition(),
+                                Map.Entry::getValue
+                        )
+                );
+    }
+
+    /**
+     * Find current offset for each partition on topic being consumed by consumer group.
+     *
+     * <p>
+     * If the consumer group does not exist, the returned map will be empty.
+     * If the consumer group exists, but it is not consuming the topic, the returned map will be empty.
+     *
+     * @param groupId consumer group ID
+     * @param topic topic the group is consuming
+     * @return map of partition to offset
+     */
+    public Map<Integer, Long> getGroupOffsets(final String groupId, final String topic) {
+        return JavaConverters.mapAsJavaMap(adminClient.listGroupOffsets(groupId))
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getKey().topic().equals(topic))
+                .collect(
+                        Collectors.toMap(
+                                entry -> entry.getKey().partition(),
+                                entry -> (Long) entry.getValue()
+                        )
+                );
+    }
+
+    @SuppressWarnings("PMD.SystemPrintln")
+    public static void main(final String[] args) {
+        final String brokerList = args[0];
+        final String groupId = args[1];
+        final String topic = args[2];
+        try (OffsetMonitor mon = new OffsetMonitor("ot-kafka-util-om-cli", brokerList)) {
+            final Map<Integer, Long> topicSizes = mon.getTopicSizes(topic);
+            final Map<Integer, Long> groupOffsets = mon.getGroupOffsets(groupId, topic);
+            assert topicSizes.keySet().equals(groupOffsets.keySet());
```

Comment:
generally avoid `assert` keyword, prefer`Verify.verify` instead.

Reply:
Since assert if "free" I dunno if I agree, but this isn't expensive, which is probably your point

Reply:
@stevenschlansker Why?  In this simple of a use case, what's wrong with `assert`?

Reply:
Assertions are enabled or disabled via a VM option at startup time.  So unless you explicitly go to your deployment and pass the `-ea` flag, it is effectively compiled out.

`Verify` and friends are always active.

Unless the check is very expensive, I prefer a solution that is always on, rather than a check that may or may not be helpful depending on how the user is deploying.

Reply:
FFS.  Updated in e407aad1360c77fc483028b8e5f04cf7fe6a6293.

Reply:
Sure, it's what your goal is. An argument I've heard - which I mostly disgree with given better TDD matchers is the assert even when disabled documents in situ some of your core assumptions. I think there can be some value there. @stevenschlansker  since many people may miss the test's deeper points